### PR TITLE
[mvfst ] : Add new port

### PR DESCRIPTION
**Pull request by jiayuehua**
**Describe the pull request**
mvfst (Pronounced move fast) is a client and server implementation of IETF QUIC protocol in C++ by Facebook.
- #### What does your PR fix?  
  Fixes #...

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  <linux>, <Yes>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/mvfst/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    REPO facebookincubator/mvfst
+    REF 497f6e4732bf6e95553d70e24f2e6e1a20c26397 
+    SHA512 41bb998f8183839f532c32755572df4a8d38a84ad164011675703a4de73347292b77134fe53359a5622b72c5b1f3fea4b295b8740eda754c860b41f75ba7f751
+    HEAD_REF main
+)
+
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/mvfst/portfile.cmake:14]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/mvfst/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    OPTIONS
+        -DBUILD_TESTS=OFF
+        -DBUILD_EXAMPLES=OFF
+)
+
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/mvfst)
+file(REMOVE_RECURSE
+${CURRENT_PACKAGES_DIR}/debug/include
+)
```
Comment (JonLiu1993) [inline ports/mvfst/portfile.cmake:24]:
```suggestion
vcpkg_cmake_install()
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/mvfst)
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```

File: ports/mvfst/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "mvfst",
+  "version-string": "0.1",
```
Comment (JonLiu1993) [inline ports/mvfst/vcpkg.json:3]:
```suggestion
  "version": "0.1",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@jiayuehua ,Thanks for your pr, After finished this review suggestion, please run command "./vcpkg x-add-version mvfst --overwrite-version" and commit again

**Comment by jiayuehua** [comment]
> @jiayuehua ,Thanks for your pr, After finished this review suggestion, please run command "./vcpkg x-add-version mvfst --overwrite-version" and commit again

ok, I will fix it as soon as possible

File: ports/mvfst/portfile.cmake (hunk 1 above)
Comment (JonLiu1993) [reply ports/mvfst/portfile.cmake]: @jiayuehua ,Please apply this review suggesstion

File: ports/mvfst/portfile.cmake (hunk 2 above)
Comment (JonLiu1993) [reply ports/mvfst/portfile.cmake]: Same as above

File: ports/mvfst/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO facebookincubator/mvfst
+    REF 497f6e4732bf6e95553d70e24f2e6e1a20c26397 
+    SHA512 41bb998f8183839f532c32755572df4a8d38a84ad164011675703a4de73347292b77134fe53359a5622b72c5b1f3fea4b295b8740eda754c860b41f75ba7f751
+    HEAD_REF main
+)
+
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
```
Comment (JonLiu1993) [inline ports/mvfst/portfile.cmake:13]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

Comment (JonLiu1993) [reply ports/mvfst/portfile.cmake:13]: @jiayuehua ,Please add double quotes to these absolute paths.

File: ports/mvfst/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    OPTIONS
+        -DBUILD_TESTS=OFF
+        -DBUILD_EXAMPLES=OFF
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/mvfst)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
```
Comment (JonLiu1993) [inline ports/mvfst/portfile.cmake:21]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```

File: ports/mvfst/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (JonLiu1993) [inline ports/mvfst/portfile.cmake:23]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/mvfst/vcpkg.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  "homepage": "https://github.com/facebookincubator/mvfst",
+  "supports": "!windows",
+  "dependencies": [
+    "boost",
+    "fizz",
+    "fmt",
+    "folly",
+    "glog"
+  ]
+}
```
Comment (JonLiu1993) [inline ports/mvfst/vcpkg.json:14]:
```suggestion
    "glog",
    {
        "name": "vcpkg-cmake",
        "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
  ]
}
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/mvfst/portfile.cmake (hunk 4 above)
Comment (jiayuehua) [reply ports/mvfst/portfile.cmake:13]: @JonLiu1993 , ok , I will add them as soon as possible.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Looks good, thanks for the new port!

**Pull request merged**